### PR TITLE
socket Refactor

### LINKED ISSUE #110: Move Socket Outside
**Describe the bug**
I think we should move all socket.io code outside of index.js in the backend because there is a bunch of socket.io events happening in index.js. 

**To Reproduce**

**Expected result**
We can create an object of the socket in the socket.js file which we store in Utils folder So I think it's easy to understand index.js code and also code will be refactored in proper structure

**Actual result**

**Screenshots**

**Are you willing to contribute to this issue?** Yes

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 89.0.4389.114

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by nishant23122000**
As I mentioned in issue #110, We should outside the socket io code So I created one socket.js file in the Utils folder of the backend and move all the socket code of index.js in socket.js.Review the change which I made.

### CODE REVIEW

**Comment by Vishv07** [comment]
@kothariji LGTM🚀. Although I would like to suggest some changes hence This PR is about refactoring. 

1) Putting Room.js and Socket.js inside utils doesn't make sense after all they are the core parts.
2) let's keep it Inside a new folder called socket. bcz util usually contains different kinds of stuff.
3) We can keep Utils folder or we can delete it as of now if we follow point 2 because after all, it will be an empty folder.

File: backend/server/index.js [hunk 1]
```diff
@@ -1,73 +1,21 @@
 const app = require("express")();
 const server = require("http").createServer(app);
 const cors = require("cors");
-const Rooms = require("./Utils/Rooms");
+
 const io = require("socket.io")(server, { origins: "*:*" });
 const helmet = require('helmet');
 const rateLimiter = require('./rateLimiter');
+const handleSocket = require('./Utils/Socket');
 // instantiate a new rooms object to store all clients in the room
```
Comment (Vishv07) [inline backend/server/index.js:9]: Delete this comment from here and just Put this inside a socket.js file where you are creating a room object.

Comment (nishant23122000) [reply backend/server/index.js:9]: Yes, there is no meaning to put room and socket.js file inside utils file,Ok I will change it

**Comment by kothariji** [comment]
hi @nishant23122000, can you please resolve the merge conflicts, so that we can review the PR 🙌 😊

**Pull request closed without merging**